src/main/java/org/apache/maven/plugins/deploy/DeployMojo.java

```diff
@@ -285,8 +297,17 @@ private void processProject(final MavenProject project, DeployRequest request) t
             if (isFile(mavenMainArtifact.getFile())) {
                 request.addArtifact(RepositoryUtils.toArtifact(mavenMainArtifact));
             } else if (!project.getAttachedArtifacts().isEmpty()) {
-                throw new MojoExecutionException("The packaging plugin for this project did not assign "
-                        + "a main file to the project but it has attachments. Change packaging to 'pom'.");
+                if (allowIncompleteProjects) {
+                    getLog().warn("");
+                    getLog().warn("The packaging plugin for this project did not assign");
```

Comment:
Hi @cstamas,

Since `processProject` also gets called from `deployAllAtOnce`, the wording "this project" is a bit ambiguous. In addition to that, if an exception is thrown in "deploy at once" mode, it will get attributed to the last project that is marked with a deployment state enum, which can also be misleading.

Is it possible to add the name of the method parameter `project` into the log and exceptions thrown from this mehod?

Reply:
Sure! Quality patches accepted.